pkg/controllers/subnetbinding/subnets_handler.go

```diff
@@ -0,0 +1,108 @@
+		newObj, newOK := e.ObjectNew.(*v1alpha1.Subnet)
+		if !oldOK || !newOK {
+			return false
+		}
+		return common.IsObjectUpdateToReady(oldObj.Status.Conditions, newObj.Status.Conditions) ||
+			common.IsObjectUpdateToUnready(oldObj.Status.Conditions, newObj.Status.Conditions)
+
+	},
+	DeleteFunc: func(e event.DeleteEvent) bool {
+		return false
+	},
+	GenericFunc: func(e event.GenericEvent) bool {
+		return false
+	},
+}
+
+var PredicateFuncsSubnetSets = predicate.Funcs{
+	CreateFunc: func(e event.CreateEvent) bool {
+		return false
+	},
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		oldObj, oldOK := e.ObjectOld.(*v1alpha1.SubnetSet)
+		newObj, newOK := e.ObjectNew.(*v1alpha1.SubnetSet)
+		if !oldOK || !newOK {
+			return false
+		}
+		return !reflect.DeepEqual(oldObj.Status.Subnets, newObj.Status.Subnets)
+	},
+	DeleteFunc: func(e event.DeleteEvent) bool {
+		return false
+	},
+	GenericFunc: func(e event.GenericEvent) bool {
+		return false
+	},
+}
+
+func requeueBindingMapsBySubnetUpdate(ctx context.Context, c client.Client, _, objNew client.Object, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	subnet := objNew.(*v1alpha1.Subnet)
+	requeueSubnetConnectionBindingMapsBySubnet(ctx, c, subnet.Namespace, subnet.Name, q)
+}
+
+func requeueSubnetConnectionBindingMapsBySubnet(ctx context.Context, c client.Client, namespace string, subnet string, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	bindingMapList := &v1alpha1.SubnetConnectionBindingMapList{}
+	err := c.List(ctx, bindingMapList, client.InNamespace(namespace))
+	if err != nil {
+		log.Error(err, "Failed to list SubnetConnectionBindingMaps with Subnet event", "Namespace", namespace, "Subnet", subnet)
+		return
+	}
+	for _, bm := range bindingMapList.Items {
+		if bm.Spec.SubnetName == subnet || bm.Spec.TargetSubnetName == subnet {
+			log.Info("Requeue SubnetConnectionBindingMap because the dependent Subnet realization state is changed", "Namespace", namespace, "Name", bm.Name)
+			q.Add(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      bm.Name,
+					Namespace: bm.Namespace,
+				},
+			})
+		}
+	}
+}
+
+func requeueBindingMapsBySubnetSetUpdate(ctx context.Context, c client.Client, objOld, objNew client.Object, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	subnetSet := objNew.(*v1alpha1.SubnetSet)
+	requeueSubnetConnectionBindingMapsBySubnetSet(ctx, c, subnetSet.Namespace, subnetSet.Name, q)
+}
+
+func requeueSubnetConnectionBindingMapsBySubnetSet(ctx context.Context, c client.Client, namespace string, subnetSet string, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	bindingMapList := &v1alpha1.SubnetConnectionBindingMapList{}
+	err := c.List(ctx, bindingMapList, client.InNamespace(namespace))
+	if err != nil {
+		log.Error(err, "Failed to list SubnetConnectionBindingMaps with SubnetSet event", "Namespace", namespace, "Subnet", subnetSet)
+		return
+	}
+	for _, bm := range bindingMapList.Items {
```

Comment:
maybe we can use listOpts in List API to only return items we want?
listOpts := client.ListOptions{FieldSelector: fields.OneTermEqualSelector("spec.TargetSubnetSetName", subnetSet).String()}

Reply:
Let me try

Reply:
update: We can't use `spec.TargetSubnetSetName` as a filter since we didn't make indexes on the manager with these field(s). So I revert back to the previous version. 

If we want to do it, I would suggest to use another change for the improvement.